packages/babel-plugin-transform-react-jsx/src/create-plugin.ts

```diff
@@ -160,6 +160,8 @@ export default function createPlugin({
       },
     };
 
+    let leadingComments: t.Comment[] | undefined;
```

Comment:
Personally, I think it’s a bit hacky to intertwine the `leadingComments` logic with the attribute accumulator. This approach doesn’t address how we should handle the `trailingComments` of a `JSXSpreadAttribute`, such as in `<div {...{ id: 'hello' }/*i18n*/} />`.

Could we record the source node where the comments originate and the target node to which they will be attached? Then we could call the `t.inheritsComments` API, which was designed for comment adjustment.